### PR TITLE
Replaced the deprecated package `k8s.io/utils/pointer` with kubevirt package `kubevirt.io/kubevirt/pkg/pointer` in the `/pkg/storage` directory.

### PR DESCRIPTION
**Pull request by anishbista60**
Replaced the deprecated package `k8s.io/utils/pointer` with kubevirt package `kubevirt.io/kubevirt/pkg/pointer` in the `/pkg/storage` directory.

### What this PR does
Before this PR:
- In the `/pkg/virt-api` we used `k8s.io/utils/pointer` package which is deprecated. 

After this PR:
- Now, it is replaced with kubevirt package `kubevirt.io/kubevirt/pkg/pointer`

Ref #12949

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
None
```

### CODE REVIEW

File: pkg/storage/export/export/export.go [hunk 1]
```diff
@@ -42,14 +42,14 @@ import (
...
 	"k8s.io/client-go/util/workqueue"
-	"k8s.io/utils/pointer"
-
 	virtv1 "kubevirt.io/api/core/v1"
 	exportv1 "kubevirt.io/api/export/v1beta1"
 	"kubevirt.io/client-go/kubecli"
 	"kubevirt.io/client-go/log"
 	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
 
+	"kubevirt.io/kubevirt/pkg/pointer"
```
Comment (fossedihelm) [inline pkg/storage/export/export/export.go:51]: Can you move this import in the `kubevirt.io/kubevirt` import block below?
This follows our coding style: https://github.com/kubevirt/kubevirt/blob/main/docs/coding-conventions.md

File: pkg/storage/export/export/vmsnapshot-source_test.go [hunk 2]
```diff
@@ -53,6 +51,8 @@ import (
 	"kubevirt.io/client-go/kubecli"
 	cdiv1 "kubevirt.io/containerized-data-importer-api/pkg/apis/core/v1beta1"
 
+	"kubevirt.io/kubevirt/pkg/pointer"
```
Comment (fossedihelm) [inline pkg/storage/export/export/vmsnapshot-source_test.go:54]: same as above

**Comment by anishbista60** [comment]
@fossedihelm done.

**Review by ShellyKa13 — COMMENTED** [review]
nice, thanks!
/lgtm

File: pkg/storage/export/export/export.go [hunk 3]
```diff
@@ -877,8 +876,8 @@ func (ctrl *VMExportController) createExporterPodManifest(vmExport *exportv1.Vir
 	podManifest.Labels = map[string]string{exportServiceLabel: vmExport.Name}
 	podManifest.Annotations = map[string]string{annCertParams: scp}
 	podManifest.Spec.SecurityContext = &corev1.PodSecurityContext{
-		RunAsNonRoot:   pointer.Bool(true),
-		FSGroup:        pointer.Int64Ptr(kvm),
+		RunAsNonRoot:   pointer.P(true),
+		FSGroup:        pointer.P(int64(kvm)),
```
Comment (iholder101) [inline pkg/storage/export/export/export.go:880]: nit: AFAICT this is the only usage of `kvm`, maybe it's better to change it to `int64`?

**Review by iholder101 — COMMENTED** [review]
Thank you @anishbista60!
/approve

File: pkg/storage/export/export/export.go (hunk 3 above)
Comment (anishbista60) [reply pkg/storage/export/export/export.go:880]: @iholder101 yes, used in only one place. Now, what to do ?. shall i open for new comers?

Comment (iholder101) [reply pkg/storage/export/export/export.go:880]: It's just a nit (=not important), I don't think it's crucial in any sense :)

Comment (anishbista60) [reply pkg/storage/export/export/export.go:880]: ok

**Pull request merged**